crates/pop-contracts/src/templates.rs

```diff
@@ -0,0 +1,244 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use crate::errors::Error;
+use strum::{
+	EnumMessage as EnumMessageT, EnumProperty as EnumPropertyT, VariantArray as VariantArrayT,
+};
+use strum_macros::{AsRefStr, Display, EnumMessage, EnumProperty, EnumString, VariantArray};
+
+/// Supported template providers.
+#[derive(
+	AsRefStr, Clone, Default, Debug, Display, EnumMessage, EnumString, Eq, PartialEq, VariantArray,
+)]
+pub enum ContractType {
+	#[default]
+	#[strum(
+		ascii_case_insensitive,
+		serialize = "examples",
+		message = "Examples",
+		detailed_message = "Contract examples for ink!."
+	)]
+	Examples,
+	#[strum(
+		ascii_case_insensitive,
+		serialize = "erc",
+		message = "ERC",
+		detailed_message = "ERC-based contracts in ink!."
+	)]
+	Erc,
+}
+
+impl ContractType {
+	/// Get the list of providers supported.
+	pub fn types() -> &'static [ContractType] {
+		ContractType::VARIANTS
+	}
+
+	/// Get provider's name.
+	pub fn name(&self) -> &str {
+		self.get_message().unwrap_or_default()
+	}
+
+	/// Get the default template of the provider.
+	pub fn default_type(&self) -> Template {
+		match &self {
+			ContractType::Examples => Template::Standard,
+			ContractType::Erc => Template::ERC20,
+		}
+	}
+
+	/// Get the providers detailed description message.
+	pub fn description(&self) -> &str {
+		self.get_detailed_message().unwrap_or_default()
+	}
+
+	/// Get the list of templates of the provider.
+	pub fn templates(&self) -> Vec<&Template> {
+		Template::VARIANTS
+			.iter()
+			.filter(|t| t.get_str("ContractType") == Some(self.name()))
+			.collect()
+	}
+}
+
+#[derive(
+	AsRefStr,
+	Clone,
+	Debug,
+	Default,
+	Display,
+	EnumMessage,
+	EnumProperty,
+	EnumString,
+	Eq,
+	Hash,
+	PartialEq,
+	VariantArray,
+)]
+pub enum Template {
+	/// A minimalist contract template.
+	#[default]
+	#[strum(
+		serialize = "standard",
+		message = "Standard",
+		detailed_message = "ink!'s 'Hello World': Flipper",
+		props(ContractType = "Examples")
+	)]
+	Standard,
+	/// The implementation of the ERC-20 standard in ink!
+	#[strum(
+		serialize = "erc20",
+		message = "Erc20",
```

Comment:
```suggestion
		message = "ERC-20",
```

Reply:
Got the name of the contract from here: https://github.com/use-ink/ink-examples/blob/main/erc20/lib.rs#L10 And used this message to replace the name in the contract once cloned.